py/server/deephaven/multijoin.py

```diff
@@ -0,0 +1,126 @@
+
+        Raises:
+            DHError
+        """
+        try:
+            self.table = table
+            on = to_sequence(on)
+            joins = to_sequence(joins)
+            self.j_multijoininput = _JMultiJoinInput.of(table.j_table, on, joins)
+        except Exception as e:
+            raise DHError(e, "failed to build a MultiJoinInput object.") from e
+
+
+class MultiJoinTable(JObjectWrapper):
+    """A MultiJoinTable represents the result of a multiJoin. """
+    j_object_type = _JMultiJoinTable
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_multijointable
+
+    def table(self) -> Table:
+        return Table(j_table=self.j_multijointable.table())
+
+    def __init__(self, input: Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]],
+                 on: Union[str, Sequence[str]] = None):
+        """Creates a new MultiJoinTable from MultiJoinInput or directly from Tables and key columns. Either
+        MultiJoinInput objects or Table objects should be provided as the `input` parameter.
+
+        Args:
+            input (Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]]): the input objects
+                specifying the tables and columns to include in the join.
+            on (Union[str, Sequence[str]], optional): the column(s) to match if Table objects are provided as the input
+                parameter, must be a common name or an equality expression that matches every input table, i.e.
+                "col_a = col_b" to rename output column names.  When using MultiJoinInput objects, this parameter
+                is ignored.
+
+        Raises:
+            DHError
+        """
+        try:
+            if isinstance(input, Table) or (isinstance(input, Sequence) and all(isinstance(t, Table) for t in input)):
+                tables = to_sequence(input, wrapped=True)
+                with auto_locking_ctx(*tables):
+                    j_tables = to_sequence(input)
+                    self.j_multijointable = _JMultiJoinFactory.of(on, *j_tables)
+            elif isinstance(input, MultiJoinInput) or (isinstance(input, Sequence) and all(isinstance(ji, MultiJoinInput) for ji in input)):
+                wrapped_input = to_sequence(input, wrapped=True)
+                tables = [ji.table for ji in wrapped_input]
+                with auto_locking_ctx(*tables):
+                    input = to_sequence(input)
+                    self.j_multijointable = _JMultiJoinFactory.of(*input)
+            else:
+                raise DHError(message="input must be Table or MultiJoinInput.")
+
+        except Exception as e:
+            raise DHError(e, "failed to build a MultiJoinTable object.") from e
+
+
+"""Creates a new MultiJoinTable from MultiJoinInput or directly from tables and key columns. Either
+MultiJoinInput objects or Table objects should be provided as the `input` parameter. 
```

Comment:
Model the docstring off of https://deephaven.io/core/pydoc/code/deephaven.table.html#deephaven.table.Table.natural_join